src/order/flag.lean

```diff
@@ -0,0 +1,99 @@
+/-
+Copyright (c) 2023 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import data.set.pointwise.smul
+import order.chain
+import order.grade
+import order.rel_iso.group
+
+/-!
+# Additional constructions about flags
+
+This file defines the action of order isomorphisms on flags and grades the elements of a flag.
+
+## TODO
+
+The file structure doesn't seem optimal. Maybe all the `flag` material could move here, or to a
+subfolder?
+-/
+
+open_locale pointwise
+
+variables {𝕆 α : Type*}
+
+namespace flag
+
+/-!
+### Action on flags
+
+Order isomorphisms act on flags.
+-/
+
+section preorder
+variables [preorder α]
+
+instance : has_smul (α ≃o α) (flag α) :=
+⟨λ e s,
+  { carrier := e • s,
+    chain' := s.chain_le.image _ _ _ e.monotone,
+    max_chain' := λ t ht hst, (smul_eq_iff_eq_inv_smul _).2 $ s.max_chain.2
+      (ht.image _ _ _ e.symm.monotone) $ set.set_smul_subset_iff.1 hst }⟩
```

Comment:
Is there a `flag.map` hiding here that works for `α ≃o b`?

Reply:
Well spotted!